File: ReSharper.FSharp/src/Psi.FSharp/src/FSharpPsiProperties.cs
using JetBrains.Annotations;
using JetBrains.Platform.ProjectModel.FSharp;
using JetBrains.ProjectModel;
using JetBrains.ReSharper.Psi.Impl;

namespace JetBrains.ReSharper.Psi.FSharp
{
  public class FSharpPsiProperties : DefaultPsiProjectFileProperties
  {
    public FSharpPsiProperties([NotNull] IProjectFile projectFile, [NotNull] IPsiSourceFile sourceFile)
      : base(projectFile, sourceFile)
    {
    }

    public override bool ProvidesCodeModel =>
      ProjectFile.Properties.BuildAction.IsCompile() ||
      ProjectFile.LanguageType.Equals(FSharpScriptProjectFileType.Instance);

    public override bool ShouldBuildPsi => true;
  }
}

File: ReSharper.FSharp/src/FSharp.Psi.Features/src/LanguageService/FSharpPsiProperties.fs
namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.LanguageService

open JetBrains.ReSharper.Psi.Impl

type FSharpPsiProperties(projectFile, sourceFile, providesCodeModel) =
    inherit DefaultPsiProjectFileProperties(projectFile, sourceFile)

    override x.ProvidesCodeModel = providesCodeModel
    override x.ShouldBuildPsi = true
